src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -347,15 +347,23 @@ class IMain(val settings: Settings, parentClassLoaderOverride: Option[ClassLoade
   override def translateEnclosingClass(n: String): Option[String] = symbolOfTerm(n).enclClass.toOption map flatPath
 
   /** If unable to find a resource foo.class, try taking foo as a symbol in scope
-    *  and use its java class name as a resource to load.
-    *
-    *  \$intp.classLoader classBytes "Bippy" or \$intp.classLoader getResource "Bippy.class" just work.
-    */
+   *  and use its java class name as a resource to load.
+   *
+   *  \$intp.classLoader classBytes "Bippy" or \$intp.classLoader getResource "Bippy.class" just work.
+   */
   private class TranslatingClassLoader(parent: ClassLoader) extends AbstractFileClassLoader(replOutput.dir, parent) {
     override protected def findAbstractFile(name: String): AbstractFile = super.findAbstractFile(name) match {
       case null if _initializeComplete => translateSimpleResource(name).map(super.findAbstractFile).orNull
       case file => file
     }
+    // if the name was mapped by findAbstractFile, supply null name to avoid name check in defineClass
+    override protected def findClass(name: String): Class[_] = {
```

Comment:
do `super.findClass` first and use the new code below only as a fallback?

Reply:
I considered doing something fancy, but the class loader API is so complex already, plus `ScalaClassLoader` and `AbstractFileClassLoader`, and also throws, that I wasn't "up" for it.

I could imagine a boolean member called `verifyDefiningName` for `AbstractFileClassLoader` to consult.

Reply:
A complex mix of complex ingredients, so I'm just generally worried about breaking stuff... I just saw now the `AbstractFileClassLoader.findClass` code that's overridden. In that light, LGTM.